solutions/fully-configurable/main.tf

```diff
@@ -0,0 +1,282 @@
+  ssh_key_ids                      = local.ssh_keys
+  machine_type                     = var.machine_type
+  vsi_per_subnet                   = 1
+  user_data                        = var.user_data
+  skip_iam_authorization_policy    = local.create_cross_account_auth_policy ? false : var.skip_block_storage_kms_iam_auth_policy
+  boot_volume_encryption_key       = local.boot_volume_kms_key_crn
+  use_boot_volume_key_as_default   = var.use_boot_volume_key_as_default
+  kms_encryption_enabled           = var.kms_encryption_enabled_boot_volume
+  manage_reserved_ips              = var.manage_reserved_ips
+  use_static_boot_volume_name      = var.use_static_boot_volume_name
+  enable_floating_ip               = var.enable_floating_ip
+  allow_ip_spoofing                = var.allow_ip_spoofing
+  create_security_group            = var.security_group != null ? true : false
+  security_group                   = var.security_group
+  security_group_ids               = var.security_group_ids
+  block_storage_volumes            = var.block_storage_volumes
+  load_balancers                   = var.load_balancers
+  access_tags                      = var.vsi_access_tags
+  snapshot_consistency_group_id    = var.snapshot_consistency_group_id
+  boot_volume_snapshot_id          = var.boot_volume_snapshot_id
+  enable_dedicated_host            = var.dedicated_host_id != null ? true : false
+  dedicated_host_id                = var.dedicated_host_id
+  use_legacy_network_interface     = false
+  secondary_allow_ip_spoofing      = var.secondary_allow_ip_spoofing
+  secondary_floating_ips           = var.secondary_floating_ips
+  secondary_security_groups        = var.secondary_security_groups
+  secondary_use_vsi_security_group = var.secondary_use_vsi_security_group
+  secondary_subnets                = local.secondary_subnet
+  placement_group_id               = var.placement_group_id
+  primary_vni_additional_ip_count  = var.primary_virtual_network_interface_additional_ip_count
+  custom_vsi_volume_names          = local.custom_vsi_volume_names
+}
+
+########################################################################################################################
+# Secret Mananger
+########################################################################################################################
+
+module "existing_secret_manager_crn_parser" {
+  count   = var.existing_secrets_manager_instance_crn != null ? 1 : 0
+  source  = "terraform-ibm-modules/common-utilities/ibm//modules/crn-parser"
+  version = "1.1.0"
+  crn     = var.existing_secrets_manager_instance_crn
+}
+locals {
+  existing_secrets_manager_instance_guid   = var.existing_secrets_manager_instance_crn != null ? module.existing_secret_manager_crn_parser[0].service_instance : null
+  existing_secrets_manager_instance_region = var.existing_secrets_manager_instance_crn != null ? module.existing_secret_manager_crn_parser[0].region : null
+}
+
+module "secrets_manager_service_credentials" {
```

Comment:
Hmm we are not creating a service credential (we are adding sshkey as an arbitrary secret), so I would rename this to avoid confusion.